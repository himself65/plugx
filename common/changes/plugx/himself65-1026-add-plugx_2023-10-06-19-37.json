{
  "changes": [
    {
      "packageName": "plugx",
      "comment": "",
      "type": "none"
    }
  ],
  "packageName": "plugx"
}