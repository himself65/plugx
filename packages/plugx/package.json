{
  "name": "plugx",
  "description": "Universal plugin framework for web application",
  "version": "1.0.1",
  "types": "module",
  "license": "MIT",
  "author": "Alex Yang <himself65@outlook.com>",
  "publishConfig": {
    "access": "public"
  },
  "files": [
    "index.js"
  ],
  "scripts": {
    "build": "echo 'no build currently'"
  }
}
