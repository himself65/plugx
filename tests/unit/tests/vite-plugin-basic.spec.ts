import { describe, expect, it } from 'vitest'
import { build } from 'vite'
import { fileURLToPath } from 'node:url'
import plugx from '@plugxjs/vite-plugin'
import type { RollupOutput } from 'rollup'
import type { OutputAsset, OutputChunk } from 'rollup'

describe('basic', () => {
  it('should works', async () => {
    for (const ext of ['.ts', '.js']) {
      const [{ output }] = await build({
        build: {
          minify: false,
          lib: {
            entry: fileURLToPath(
              new URL(`./fixtures/basic${ext}`, import.meta.url)),
            fileName: 'test',
            formats: ['es']
          }
        },
        plugins: [plugx()]
      }) as RollupOutput[]
      expect(output.length).toBe(2)
      const main = output[0] as OutputChunk
      const staticJson = output[1] as OutputAsset
      expect(main.fileName).toBe('test.js')
      expect(staticJson.fileName).toBe('test.js.static.json')
      expect(main.code).toMatchSnapshot()
      expect(staticJson.source).toMatchSnapshot()
    }
  })
})
