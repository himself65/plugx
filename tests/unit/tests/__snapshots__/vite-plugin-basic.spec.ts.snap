// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`basic > should works 1`] = `
"({ imports: $h‍_imports, liveVar: $h‍_live, onceVar: $h‍_once, importMeta: $h‍____meta }) => {
  $h‍_imports([]);
  console.log(\\"Hello world!\\");
};
"
`;

exports[`basic > should works 2`] = `
"{
  \\"exports\\": {},
  \\"imports\\": [],
  \\"reexports\\": {}
}"
`;

exports[`basic > should works 3`] = `
"({ imports: $h‍_imports, liveVar: $h‍_live, onceVar: $h‍_once, importMeta: $h‍____meta }) => {
  $h‍_imports([]);
  console.log(\\"Hello world!\\");
};
"
`;

exports[`basic > should works 4`] = `
"{
  \\"exports\\": {},
  \\"imports\\": [],
  \\"reexports\\": {}
}"
`;
